src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -973,6 +973,12 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 			!lastPart ||
 			lastPart.kind === 'references' ||
 			(lastPart.kind === 'toolInvocation' && (lastPart.isComplete || lastPart.presentation === 'hidden')) ||
+			// --- Start Positron ---
+			// Show working progress if the last part is markdown content and the response is in progress
+			// This is especially helpful when running in Agent mode, when the executeCode code block is still being constructed
+			// This causes the in progress indicator to show in any mode, while markdown text is being rendered and the response is not complete
+			(lastPart.kind === 'markdownContent' && element.model.isInProgress.get()) ||
```

Comment:
I noticed that with `model.isInProgress` the progress notification disappears ever so sooner than the response is finished. Using `!element.isComplete` *seems* to line the timing up better. But, it may be in my head! 
```suggestion
			(lastPart.kind === 'markdownContent' && !element.isComplete) ||
```

Reply:
What prompt and chat mode did you use? I am seeing similar timing, but happy to switch to this as it's shorter :)

`isInProgress` has a couple other conditions, but shouldn't come into play for usual prompts:

https://github.com/posit-dev/positron/blob/932a61ef3efe2101c4bb7638378a43a433785b51/src/vs/workbench/contrib/chat/common/chatModel.ts#L771-L778

Reply:
I followed the ask mode prompt from your sample video 🤔 Again, maybe in my head, maybe a difference in machine/internet/cosmic entropy!